This code-base is so old. This was fixed in qemu with 2bff4b6 the 23rd of
December 2009:

  Compile qemu-nbd also on OpenBSD and Solaris

	basename() needs #include <libgen.h>.

	No prototype for daemon() is available on Solaris, but link
	succeeds.

--- a/tools/qemu-xen-traditional/qemu-nbd.c
+++ b/tools/qemu-xen-traditional/qemu-nbd.c
@@ -31,6 +31,7 @@
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <signal.h>
+#include <libgen.h>
 
 #define SOCKET_PATH    "/var/lock/qemu-nbd-%s"
 
